cleanup.sh

```diff
@@ -271,30 +271,43 @@ for APISERVICE in $(kubectl  get apiservice -o name | grep cattle | grep -v k3s\
 done
 
 # Pod security policies
-# Rancher logging
-for PSP in $(kubectl get podsecuritypolicy -o name -l app.kubernetes.io/name=rancher-logging) podsecuritypolicy.policy/rancher-logging-rke-aggregator; do
-  kcd "$PSP"
-done
+# Get the Kubernetes server version using kubectl
+kube_version=$(kubectl version --short | awk '/Server Version:/ {print $3}')
 
-# Rancher monitoring
-for PSP in $(kubectl  get podsecuritypolicy -o name -l release=rancher-monitoring) $(kubectl get podsecuritypolicy -o name -l app=rancher-monitoring-crd-manager) $(kubectl get podsecuritypolicy -o name -l app=rancher-monitoring-patch-sa) $(kubectl get podsecuritypolicy -o name -l app.kubernetes.io/instance=rancher-monitoring); do
-  kcd "$PSP"
-done
+# Define the target Kubernetes version
+target_version="v1.25.0"
 
-# Rancher OPA
-for PSP in $(kubectl  get podsecuritypolicy -o name -l release=rancher-gatekeeper) $(kubectl get podsecuritypolicy -o name -l app=rancher-gatekeeper-crd-manager); do
-  kcd "$PSP"
-done
+# Compare the Kubernetes version with the target version
+if [[ "$kube_version" < "$target_version" ]]; then
```

Comment:
I'm pretty sure this ends up doing a string comparison, which might have unexpected results with semver strings. These failures are admittedly _well_ into the future, but it would still be nice to handle them more robustly:
```
[nflynt@reploid ~]$ kube_version="v1.200.0"
[nflynt@reploid ~]$ target_version="v1.25.0"
[nflynt@reploid ~]$ if [[ "$kube_version" < "$target_version" ]]; then echo "yes"; fi
yes
[nflynt@reploid ~]$
```

```
[nflynt@reploid ~]$ kube_version="v1.25.21"
[nflynt@reploid ~]$ target_version="v1.25.200"
[nflynt@reploid ~]$ if [[ "$kube_version" < "$target_version" ]]; then echo "yes"; fi
[nflynt@reploid ~]$ 
```

Reply:
I feel like they'd go up to 2.0.0 before 1.200.0 but yea I see what you mean.

Reply:
Switched conditional to check if psps exist through kubectl.
```
kubectl get podsecuritypolicy > /dev/null 2>&1

# Check the exit code of the previous command
if [ $? -ne 0 ]; then
  echo "PSPs deprecated in Kubernetes version $kube_version"
else
  echo "Removing PSPs"
fi
```